src/Makefile

```diff
@@ -352,41 +353,39 @@ GRPC_GEN_OBJS:= \
 	cartesi-machine-checkin.grpc.pb.o
 
 LIBCARTESI_GRPC_OBJS:= \
-	sha3.o \
-	machine-merkle-tree.o \
-	pristine-merkle-tree.o \
-	$(GRPC_GEN_OBJS) \
-	$(PROTOBUF_GEN_OBJS) \
 	grpc-virtual-machine.o \
+	grpc-machine-c-api.o \
 	protobuf-util.o \
-	grpc-machine-c-api.o
+	$(GRPC_GEN_OBJS) \
+	$(PROTOBUF_GEN_OBJS) \
+	$(LIBCARTESI_OBJS)
+
+LIBCARTESI_JSONRPC_OBJS:= \
+	jsonrpc-virtual-machine.o \
+	jsonrpc-machine-c-api.o \
+	mongoose.o \
+	$(LIBCARTESI_OBJS)
 
 LUACARTESI_GRPC_OBJS:= \
-	clua-i-virtual-machine.o \
-	clua-htif.o \
-	clua-machine-util.o \
 	clua-cartesi-grpc.o \
 	clua-grpc-machine.o \
-	clua.o
+	$(CARTESI_CLUA_OBJS) \
+	$(LIBCARTESI_GRPC_OBJS)
 
 LUACARTESI_JSONRPC_OBJS:= \
-	clua-i-virtual-machine.o \
-	clua-htif.o \
-	clua-machine-util.o \
 	clua-cartesi-jsonrpc.o \
 	clua-jsonrpc-machine.o \
-	json-util.o \
-	base64.o \
-	mongoose.o \
-	jsonrpc-machine-c-api.o \
-	jsonrpc-virtual-machine.o \
-	clua.o
+	$(CARTESI_CLUA_OBJS) \
+	$(LIBCARTESI_JSONRPC_OBJS)
```

Comment:
Shouldn´t LUACARTESI_JSONRPC simply use LIBCARTESI_JSONRPC library, instead of the objects?